### ISSUE TITLE
space-infix-ops fails with vue and typescript

### ISSUE THREAD

**Issue by KaelWD** [eslint/eslint#10260]
**Tell us about your environment**

* **ESLint Version:** 4.19.0
* **Node Version:** 8.9.4
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
typescript-eslint-parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  root: true,
  parserOptions: {
    parser: require.resolve('typescript-eslint-parser'),
    ecmaVersion: '2017',
    sourceType: 'module'
  },
  env: {
    browser: true,
  },
  extends: [
    'standard',
    'plugin:vue/recommended'
  ],
  plugins: [
    'typescript',
    'vue'
  ],
  // add your custom rules here
  rules: {
    // https://github.com/eslint/typescript-eslint-parser#known-issues
    'no-undef': 'off',
    'no-unused-vars': 'off',
    // https://github.com/eslint/typescript-eslint-parser/issues/445
    // 'typescript/no-unused-vars': 'error'
    'vue/require-default-prop': 'off',
    'no-console': 'off',
    // allow debugger during development
    'no-debugger': process.env.NODE_ENV === 'production' ? 2 : 0
  },
  overrides: [
    {
      files: '**/*.vue',
      rules: {
        indent: false,
        'vue/script-indent': ['error', 2, {
          baseIndent: 1,
          switchCase: 0
        }],
        'vue/max-attributes-per-line': [2, {
          singleline: 5,
          multiline: {
            max: 1,
            allowFirstLine: false
          }
        }],
      }
    },
  ]
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```vue
<script lang="ts">
  import Vue from 'vue'

  function foo (bar: string): string {
    return bar
  }

  export default Vue.extend({
    created () {
      let baz: any = ''
    }
  })
</script>
```

```bash
eslint --ext .ts,.js,.vue src
```
It only happens in a .vue file, the same code works fine in a .ts
Removing the assignment fixes it (so just `let baz: any`)

**What did you expect to happen?**
No errors

**What actually happened? Please include the actual, raw output from ESLint.**
```
src/components/Test.vue
  4:20  error  Infix operators must be spaced  space-infix-ops

✖ 1 problem (1 error, 0 warnings)
  1 error, 0 warnings potentially fixable with the `--fix` option.
```
Error is at the `:` in `bar: string`, but if I add a space then it moves to the next one.

**Comment by kaicataldo** [comment]
Apologies for losing track of this. Is this still an issue?

**Comment by Antipyrine101** [comment]
i have the same problem as well, what bother me is i have no infix operators in the line the error occurs，my config are:
"eslint": "5.2.0", 
"typescript": "3.0.3"、
"typescript-eslint-parser": "18.0.0"、 
"eslint-plugin-typescript": "0.12.0"、
my eslintrc is:
`
module.exports = {
  parserOptions: {
    parser: 'typescript-eslint-parser',
    ecmaVersion: 2018,
    sourceType: 'module',
    ecmaFeatures: {
      globalReturn: false,
      jsx: true
    }
  },
  env: {
    browser: true,
    node: true,
    commonjs: true,
    'shared-node-browser': true,
    worker: true,
    serviceworker: true
  },
  globals: {},

  plugins: ['typescript'],
  extends: ['standard', 'plugin:vue/recommended'],
  root: true,
  rules: {
    'vue/html-indent': [2, 4],
    'vue/html-self-closing': 0
  }
}
`
but where the error tell is like this:
`import { getPlugins, Children } from '@/pages/dashboard/util.ts'`
` groupMap: [],`
`components: {
    VPlugin
  },`
i am confused about this problem,because i have no idea if { or [ is an infix operator and  how to fix it

**Comment by ilyavolodin** [comment]
This could be an issue with `typescript-eslint-parser`. Not all of the rules are supported when using that parser, so it might be reporting something that's triggering `space-infix-ops` rule.

**Comment by KaelWD** [comment]
https://github.com/mysticatea/vue-eslint-parser/pull/31#issuecomment-412335391

I don't think there's anything actionable for this repository.
